### ISSUE TITLE
[doc] Use of the foreign libraries to build the docs should be consistent between the library and the applications.

### ISSUE THREAD

**Issue by krivenko** [TRIQS/triqs#236]
Hello,
During an iteration on the cthyb docs, I noticed that Sphinx uses different versions of extensions for the library on one side, and for cthyb on the other. As a result, `plot_directive` extension behaves differently in these two cases with very similar configuration files (it fails to produce hires/pdf illustrations for cthyb).

The extensions used for the library are those from `foreignlibs` dir. While for the applications Sphinx has to import them from somewhere else (it merely has no access to `foreignlibs`). In my opinion, this issue should be resolved either by removing  `foreignlibs` altogether, or by making them accessible to the application build system. I personally prefer the first option, as long as it does not introduce too many new dependencies.

**Comment by parcollet** [comment]
That was introduced when plot_directive was not so widely available. We should consider indeed your first option.

**Comment by krivenko** [comment]
I could go through `foreignlibs` and check what Sphinx extensions are available in Ubuntu 14.04/15.04 and what the corresponding package names are. [Here](http://ipht.cea.fr/triqs/1.2/installation/requirements.html#requirements) we have a table of prerequisites. I suggest adding one more table with dependencies needed only to build the docs. AFAIK, this is normal practice; some tiny libraries require LaTeX to build their documentation.

**Comment by parcollet** [comment]
Ok.
I agree, the deps of the doc building are not an issue. Most people do not build the doc locally...

**Comment by parcollet** [comment]
triqs_example is in foreign_libs, but it is our code ... it should be moved away, it is used in the doc build
(it is a little sphinx extension).

**Comment by parcollet** [comment]
For the rest, I would remove them if it is ok when building the doc now on Ubuntu 14.04 + some pip install if needed.
How are you doing on this ?

**Comment by krivenko** [comment]
I made the first attempt to remove `foreignlibs` (except `triqs_example`, of course, which I moved to `doc/`)...
It seems not that easy to make the documentation compile with the system-provided Sphinx extensions. Since this issue is not super-urgent I decided to postpone its solution to the end of August.

**Comment by parcollet** [comment]
What was the pb with sphinx ?

**Comment by krivenko** [comment]
I found one of the extensions (`autorun`?) to be broken. Or maybe it requires a change to the configuration file...
Another problem is that some extensions (at least `plot_directive.py`) are provided by more than one Ubuntu packages. And those versions are not identical.

**Comment by parcollet** [comment]
So we can keep our current solution.

**Comment by krivenko** [comment]
> So we can keep our current solution.

Yes, as I'm now convinced, this it is the best solution. Easy to maintain, and also easy for the people trying to build the documentation locally.

I moved the Sphinx extensions from `/foreignlibs` to `doc/sphinxext` and made them install under `${CMAKE_INSTALL_PREFIX}/share/triqs/sphinxext`, so that we can use them in cthyb and other applications. At first, I was considering putting the extensions into the main Python tree. But they are not parts of the library itself, so I decided to keep them outside.
